From d79182bb9ca0e70eda49e58d91f63177f38c463e Mon Sep 17 00:00:00 2001
From: Soeren Grunewald <soeren.grunewald@avionic-design.de>
Date: Wed, 28 Mar 2012 09:44:49 +0200
Subject: [PATCH 1/3] ARM: Silence the build on arm.

This will disable the -Wcast-align warnings appearing on arm on almost every
file to build.
It also fixes the a warning caused by uninitialized class members.

Signed-off-by: Soeren Grunewald <soeren.grunewald@avionic-design.de>
---
 GNUmakefile.am                                     |    2 +-
 .../properties/SVGAnimatedPropertyDescription.h    |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/GNUmakefile.am b/GNUmakefile.am
index bb21631..8355cda 100644
--- a/GNUmakefile.am
+++ b/GNUmakefile.am
@@ -119,7 +119,7 @@ pkgconfig_DATA :=
 %:: SCCS/s.%
 
 global_cppflags += \
-	-Wall -W -Wcast-align -Wchar-subscripts -Wreturn-type \
+	-Wall -W -Wchar-subscripts -Wreturn-type \
 	-Wformat -Wformat-security -Wno-format-y2k -Wundef \
 	-Wmissing-format-attribute -Wpointer-arith -Wwrite-strings \
 	-Wno-unused-parameter -Wno-parentheses \
diff --git a/Source/WebCore/svg/properties/SVGAnimatedPropertyDescription.h b/Source/WebCore/svg/properties/SVGAnimatedPropertyDescription.h
index 610fb14..8a36740 100644
--- a/Source/WebCore/svg/properties/SVGAnimatedPropertyDescription.h
+++ b/Source/WebCore/svg/properties/SVGAnimatedPropertyDescription.h
@@ -41,6 +41,7 @@ struct SVGAnimatedPropertyDescription {
     // Deleted value
     SVGAnimatedPropertyDescription(WTF::HashTableDeletedValueType)
         : m_element(reinterpret_cast<SVGElement*>(-1))
+        , m_attributeName(0)
     {
     }
 
diff --git a/Source/JavaScriptCore/assembler/MacroAssemblerARM.h
b/Source/JavaScriptCore/assembler/MacroAssemblerARM.h
index 5117389..d1143e8 100644
--- a/Source/JavaScriptCore/assembler/MacroAssemblerARM.h
+++ b/Source/JavaScriptCore/assembler/MacroAssemblerARM.h
@@ -443,9 +443,8 @@ public:
             m_assembler.ldr_un_imm(ARMRegisters::S0, right.m_value);
             m_assembler.cmp_r(left, ARMRegisters::S0);
         } else {
-            ARMWord tmp = (right.m_value == 0x80000000) ? ARMAssembler::INVALID_IMM : m_assembler.getOp2(-right.m_value);
-            if (tmp != ARMAssembler::INVALID_IMM)
-                m_assembler.cmn_r(left, tmp);
+            if ((uint32_t)right.m_value == 0x80000000)
+                m_assembler.cmn_r(left, m_assembler.getOp2(-right.m_value));
             else
                 m_assembler.cmp_r(left, m_assembler.getImm(right.m_value, ARMRegisters::S0));
         }
-- 
1.7.9.3

